packages/assets-controllers/src/TokensController.ts

```diff
@@ -263,34 +270,56 @@ export class TokensController extends BaseController<
     });
   }
 
-  _instantiateNewEthersProvider(): any {
-    return new Web3Provider(this.config?.provider);
+  /**
+   * Get the relevant provider instance.
+   *
+   * @param networkClientId - Network Client ID.
+   * @returns Web3Provider instance.
+   */
+  getProvider(networkClientId?: NetworkClientId): Web3Provider {
+    const provider = networkClientId
+      ? this.getNetworkClientById(networkClientId).provider
+      : this.config?.provider;
+
+    return new Web3Provider(provider);
   }
 
   /**
    * Adds a token to the stored token list.
    *
-   * @param address - Hex address of the token contract.
-   * @param symbol - Symbol of the token.
-   * @param decimals - Number of decimals the token uses.
-   * @param options - Object containing name and image of the token
-   * @param options.name - Name of the token
-   * @param options.image - Image of the token
-   * @param options.interactingAddress - The address of the account to add a token to.
+   * @param options - The method options.
+   * @param options.address - Hex address of the token contract.
+   * @param options.symbol - Symbol of the token.
+   * @param options.decimals - Number of decimals the token uses.
+   * @param options.options - Object containing name and image of the token
+   * @param options.options.name - Name of the token
+   * @param options.options.image - Image of the token
+   * @param options.options.interactingAddress - The address of the account to add a token to.
+   * @param networkClientId - Network Client ID.
    * @returns Current token list.
    */
   async addToken(
-    address: string,
-    symbol: string,
-    decimals: number,
     {
-      name,
-      image,
-      interactingAddress,
-    }: { name?: string; image?: string; interactingAddress?: string } = {},
+      address,
+      symbol,
+      decimals,
+      options: { name, image, interactingAddress } = {},
+    }: {
+      address: string;
+      symbol: string;
+      decimals: number;
+      options?: { name?: string; image?: string; interactingAddress?: string };
+    },
+    networkClientId?: NetworkClientId,
```

Comment:
hmmm wondering if `networkClientId` should just be in the object? Feels a bit confusing/counterintuitive to use parameter destructuring for everything other than this one?